pkg/controller/plan/migration.go

```diff
@@ -110,6 +110,23 @@ func (r Migration) Run() (reQ time.Duration, err error) {
 		if vm.MarkedCompleted() {
 			continue
 		}
+
+		// check whether the VM has been canceled by the user
+		if r.Context.Migration.Spec.Canceled(vm.Ref) {
+			vm.SetCondition(
+				libcnd.Condition{
+					Type:     Canceled,
+					Status:   True,
+					Category: Advisory,
+					Reason:   UserRequested,
+					Message:  "The migration has been canceled.",
+					Durable:  true,
+				})
+			vm.Phase = Completed
+			log.Info("Migration [CANCELED]:", "vm", vm)
```

Comment:
I think this will be logged over-and-over for every reconcile which seems undesirable, right?

Reply:
`Cancel()` calls `vm.MarkCompleted()` after deleting the resources, which should cause that VM to be skipped on any subsequent reconcile.

Reply:
Gotcha.